torch/csrc/jit/tensorexpr/ir_deserializer.h

```diff
@@ -0,0 +1,234 @@
+  const ByteImm* deserializeByteImm(const json& json);
+
+  // void visit(const Load* v) override;
+  // void visit(const Store* v) override;
+  // void visit(const Block* v) override;
+  // void visit(const For* v) override;
+  // void visit(const Allocate* v) override;
+  // void visit(const Free* v) override;
+  // void visit(const Cond* v) override;
+  // void visit(const Term* v) override;
+  // void visit(const Polynomial* v) override;
+  // void visit(const MaxTerm* v) override;
+  // void visit(const MinTerm* v) override;
+
+  // void visit(const FunctionCall* v) override;
+  // void visit(const ReduceOp* v) override;
+
+  // void visit(const AtomicAdd* v) override;
+  // void visit(const SyncThreads* v) override;
+  // void visit(const Let* v) override;
+
+  // void visit(const Buf* v) override;
+  // const Add * deserializeAdd(const json& json);
+  // const Sub * deserializeSub(const json& json);
+  // const Mul * deserializeMul(const json& json);
+  // const Div * deserializeDiv(const json& json);
+  // const Mod * deserializeMod(const json& json);
+  // const Max * deserializeMax(const json& json);
+  // const And * deserializeAnd(const json& json);
+  // const Or * deserializeOr(const json& json);
+  // const Xor * deserializeXor(const json& json);
+  // const Lshift * deserializeLshift(const json& json);
+  // const Rshift * deserializeRshift(const json& json);
+  // const CompareSelect * deserializeCompareSelect(const json& json);
+
+  // const Expr * deserializeSub(const json& json);
+  // const Expr * deserializeMul(const json& json);
+  // const Expr * deserializeDiv(const json& json);
+  // const Expr * deserializeMod(const json& json);
+  // const Expr * deserializeMax(const json& json);
+  // const Expr * deserializeAnd(const json& json);
+  // const Expr * deserializeOr(const json& json);
+  // const Expr * deserializeXor(const json& json);
+  // const Expr * deserializeLshift(const json& json);
+  // const Expr * deserializeRshift(const json& json);
+  // const Expr * deserializeCompareSelect(const json& json);
+
+  const Expr* deserializeBinaryOp(const json& json, IRNodeType type);
+
+  //   void visit(const Add* v) override;
+  //   void visit(const Sub* v) override;
+  //   void visit(const Mul* v) override;
+  //   void visit(const Div* v) override;
+  //   void visit(const Mod* v) override;
+  //   void visit(const Max* v) override;
+  //   void visit(const Min* v) override;
+  //   void visit(const And* v) override;
+  //   void visit(const Or* v) override;
+  //   void visit(const Xor* v) override;
+  //   void visit(const Lshift* v) override;
+  //   void visit(const Rshift* v) override;
+  //   void visit(const CompareSelect* v) override;
+
+  // #define IMM_SERIALIZE_VISIT(Type, Name) void visit(const Name##Imm* v)
+  // override;
+  //   AT_FORALL_SCALAR_TYPES_AND2(Bool, Half, IMM_SERIALIZE_VISIT);
+  // #undef IMM_SERIALIZE_VISIT
+
+  //   void visit(const Cast* v) override;
```

Comment:
Once copy of these comments is definitely enough

Reply:
Lol, sorry I thought I delete all of this stuff, maybe I forgot to re-ghstack.